duneana/CAFMaker/CAFMaker_module.cc

```diff
@@ -175,299 +205,480 @@ namespace caf {
+
+      const simb::MCNeutrino &neutrino = mctruth[i].GetNeutrino();
+
+      inter.pdg = neutrino.Nu().PdgCode();
+      inter.pdgorig = flux[i].fntype;
+      inter.iscc = !(neutrino.CCNC()); // ccnc is 0=CC 1=NC
+      inter.mode = static_cast<caf::ScatteringMode>(neutrino.Mode());
+      inter.targetPDG = gtruth[i].ftgtPDG;
+      inter.hitnuc = gtruth[i].fHitNucPDG;
+      //TODO inter.removalE ; Not sure the info can be retrieved from Gtruth and MCTruth (at least not trivially)
+      inter.E = neutrino.Nu().E();
+
+      inter.vtx.SetX(neutrino.Lepton().Vx());
+      inter.vtx.SetY(neutrino.Lepton().Vy());
+      inter.vtx.SetZ(neutrino.Lepton().Vz());
+      inter.time = neutrino.Lepton().T();
+      inter.momentum.SetX(neutrino.Nu().Momentum().X());
+      inter.momentum.SetY(neutrino.Nu().Momentum().Y());
+      inter.momentum.SetZ(neutrino.Nu().Momentum().Z());
+
+      inter.W = neutrino.W();
+      inter.Q2 = neutrino.QSqr();
+      inter.bjorkenX = neutrino.X();
+      inter.inelasticity = neutrino.Y();
+
+      TLorentzVector q = neutrino.Nu().Momentum()-neutrino.Lepton().Momentum();
+      inter.q0 = q.E();
+      inter.modq = q.Vect().Mag();
+      inter.t = gtruth[i].fgT;
+      //TODO: inter.isvtxcont ; Not sure this is the best place to define containment
+
+      inter.ischarm = gtruth[i].fIsCharm;
+      inter.isseaquark = gtruth[i].fIsSeaQuark;
+      inter.resnum = gtruth[i].fResNum;
+      inter.xsec = gtruth[i].fXsec;
+      inter.genweight = gtruth[i].fweight;
+
+      //TODO: To be done later when the info will be propagated/available
+      // inter.baseline ///< Distance from decay to interaction [m]
+      // inter.prod_vtx ///< Neutrino production vertex [cm; beam coordinates]
+      // inter.parent_dcy_mom ///< Neutrino parent momentum at decay [GeV; beam coordinates]
+      // inter.parent_dcy_mode ///< Parent hadron/muon decay mode
+      // inter.parent_pdg ///< PDG Code of parent particle ID
+      // inter.parent_dcy_E ///< Neutrino parent energy at decay [GeV]
+      // inter.imp_weight ///< Importance weight from flux file
+
+      const simb::MCGeneratorInfo &genInfo = mctruth[i].GeneratorInfo();
+
+      //TODO: Ask to add all the generators in the StandardRecord
+      std::map<simb::Generator_t, caf::Generator> genMap = {
+        {simb::Generator_t::kUnknown, caf::Generator::kUnknownGenerator},
+        {simb::Generator_t::kGENIE,   caf::Generator::kGENIE},
+        {simb::Generator_t::kCRY,     caf::Generator::kUnknownGenerator},
+        {simb::Generator_t::kGIBUU,   caf::Generator::kGIBUU},
+        {simb::Generator_t::kNuWro,   caf::Generator::kUnknownGenerator},
+        {simb::Generator_t::kMARLEY,  caf::Generator::kUnknownGenerator},
+        {simb::Generator_t::kNEUT,    caf::Generator::kNEUT},
+        {simb::Generator_t::kCORSIKA, caf::Generator::kUnknownGenerator},
+        {simb::Generator_t::kGEANT,   caf::Generator::kUnknownGenerator}
+      };
+
```

Comment:
I wondered if this map could be made a member variable, so that it isn't being made afresh for each event.